tests/CSharp/CSharp.Tests.cs

```diff
@@ -834,4 +834,16 @@ public void TestGenerationOfArraySetter()
             complexArrayElement.Dispose();
         }
     }
-}
\ No newline at end of file
+
+    [Test]
+    public void TestConstRefIndexer()
+    {
+        using (var indexproperty = new TestIndexedProperties())
+        {
+            int a = 2;
+            int* b = &a;
+            int* c = &a;
+            Assert.AreEqual(indexproperty[b], indexproperty[c]);
```

Comment:
Since b and c have the same value, you compare two identical values which is pointless. Compare to a hard-coded value instead.
In addition, please use the Assert.That(... Is.EqualTo(...)) syntax because we have long decided to use it over the old one.